crates/pathfinder/src/rpc/api.rs

```diff
@@ -1277,51 +1257,75 @@ impl RpcApi {
         })
     }
 
+    /// Estimate fee on a starknet transaction locally.
     pub async fn estimate_fee(
         &self,
         request: Call,
         block_hash: BlockHashOrTag,
     ) -> RpcResult<FeeEstimate> {
         use crate::cairo::ext_py::GasPriceSource;
 
-        match (
-            self.call_handle.as_ref(),
-            self.shared_gas_price.as_ref(),
-            &block_hash,
-        ) {
-            (Some(h), _, &BlockHashOrTag::Hash(hash)) => {
-                // discussed during estimateFee work: when using block_hash use the gasPrice from
-                // the starknet_blocks::gas_price column, otherwise (tags) get the latest eth_gasPrice.
-                h.estimate_fee(request, hash.into(), GasPriceSource::PastBlock, None)
-                    .await
-                    .map_err(Error::from)
-            }
-            (Some(h), Some(g), &BlockHashOrTag::Tag(Tag::Latest)) => {
-                // now we want the gas_price from our hopefully pre-fetched source, it will be the
-                // same when we finally have pending support
+        let handle = self
+            .call_handle
+            .as_ref()
+            .ok_or_else(|| internal_server_error("Unsupported configuration"))?;
 
-                let gas_price = g
-                    .get()
+        let gas_price = if matches!(block_hash, BlockHashOrTag::Tag(Tag::Latest | Tag::Pending)) {
+            let gas_price = match self.shared_gas_price.as_ref() {
```

Comment:
Should restore the comment:

```suggestion
            // discussed during estimateFee work: when using block_hash use the gasPrice from
            // the starknet_blocks::gas_price column, otherwise (tags) get the latest eth_gasPrice.
            let gas_price = match self.shared_gas_price.as_ref() {
```

Reply:
Restored, even with added text.